packages/eas-cli/src/project/remoteVersionSource.ts

```diff
@@ -45,11 +60,21 @@ export async function ensureVersionSourceIsRemoteAsync(
   Log.withTick('Updated eas.json');
 }
 
-export function validateBuildProfileVersionSettings(
+export async function validateBuildProfileVersionSettingsAsync(
   profileInfo: ProfileData<'build'>,
-  cliConfig: EasJson['cli']
-): void {
-  if (cliConfig?.appVersionSource !== AppVersionSource.REMOTE) {
+  cliConfig: EasJson['cli'],
+  projectDir: string
+): Promise<void> {
+  if (cliConfig?.appVersionSource === undefined) {
+    const easJsonAccessor = EasJsonAccessor.fromProjectPath(projectDir);
+    const selection = await ensureAppVersionSourceIsSetAsync(easJsonAccessor);
+    if (selection === AppVersionSourceUpdateOption.SET_TO_LOCAL && cliConfig) {
+      cliConfig.appVersionSource = AppVersionSource.LOCAL;
+    } else if (selection === AppVersionSourceUpdateOption.SET_TO_REMOTE && cliConfig) {
+      cliConfig.appVersionSource = AppVersionSource.REMOTE;
+    }
```

Comment:
maybe we wouldn't need to duplicate this logic in multiple places if `await ensureAppVersionSourceIsSetAsync(easJsonAccessor)` returned the updated `cliConfig`?